connect/json/src/main/java/org/apache/kafka/connect/json/JsonConverterConfig.java

```diff
@@ -41,7 +41,7 @@
 
     public static final String SCHEMAS_INFER_ENABLE_CONFIG = "schemas.infer.enable";
     public static final boolean SCHEMAS_INFER_ENABLE_DEFAULT = false;
-    private static final String SCHEMAS_INFER_ENABLE_DOC = "Infer the schemas when they are missing within each of the serialized values and keys.";
+    private static final String SCHEMAS_INFER_ENABLE_DOC = "Infer the schemas when they are missing within each of the serialized values and keys. Only applied when " + SCHEMAS_ENABLE_CONFIG + " is false.";
```

Comment:
I might suggestion:

    "Infer the schemas when deserializing JSON and when " + SCHEMAS_ENABLE_CONFIG + " is false and the JSON keys/values have no embedded schemas."